packages/astro/src/runtime/server/jsx.ts

```diff
@@ -220,11 +220,27 @@ async function renderElement(
 		`<${tag}${spreadAttributes(props)}${markHTMLString(
 			(children == null || children == '') && voidElementNames.test(tag)
 				? `/>`
-				: `>${children == null ? '' : await renderJSX(result, children)}</${tag}>`
+				: `>${
+						children == null ? '' : await renderJSX(result, prerenderElementChildren(tag, children))
+				  }</${tag}>`
 		)}`
 	);
 }
 
+/**
+ * Pre-render the children with the given `tag` information
```

Comment:
nit nit nit: the description and/or the function's name don't do justice to the actual implementation. Saying "pre-render" is a bit reductive, considering what the function actually does. Maybe `escapeContentTags` or something like that?

Reply:
I chose `prerenderElementChildren` since it's being called by the `renderElement` function, so kinda like a pre-pass phase. So ideally in the future this function could do more than handling this specific case, similar to the other render functions in this file. So I'm not sure if we need to rename it to something specific 🤔